x/bank/client/rest/tx_test.go

```diff
@@ -0,0 +1,142 @@
+package rest_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+	types2 "github.com/cosmos/cosmos-sdk/x/auth/types"
+	rest2 "github.com/cosmos/cosmos-sdk/x/bank/client/rest"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+func (s *IntegrationTestSuite) TestCoinSend() {
+	encodingConfig := simapp.MakeEncodingConfig()
+	authclient.Codec = encodingConfig.Marshaler
+
+	val := s.network.Validators[0]
+
+	initValidatorCoins, err := getCoinsFromValidator(val)
+	s.Require().NoError(err)
+	s.Require().Equal(
+		types.NewCoins(
+			types.NewCoin(fmt.Sprintf("%stoken", val.Moniker), s.cfg.AccountTokens),
+			types.NewCoin(s.cfg.BondDenom, s.cfg.StakingTokens.Sub(s.cfg.BondedTokens)),
+		),
+		initValidatorCoins,
+	)
+
+	account, err := getAccountInfo(val)
+	s.Require().NoError(err)
+
+	sendReq := generateSendReq(
+		account,
+		types.Coins{types.NewCoin(s.cfg.BondDenom, types.TokensFromConsensusPower(1))},
+	)
+
+	stdTx, err := submitSendReq(val, sendReq)
+	s.Require().NoError(err)
+
+	s.Require().Nil(stdTx.Signatures)
+	s.Require().Equal([]types.Msg{
+		&banktypes.MsgSend{
+			FromAddress: account.GetAddress(),
+			ToAddress:   account.GetAddress(),
+			Amount:      sendReq.Amount,
+		},
+	}, stdTx.GetMsgs())
+}
+
+func submitSendReq(val *testutil.Validator, req rest2.SendReq) (types2.StdTx, error) {
+	url := fmt.Sprintf("%s/bank/accounts/%s/transfers", val.APIAddress, val.Address)
+
+	bz, err := val.ClientCtx.JSONMarshaler.MarshalJSON(req)
+	if err != nil {
+		return types2.StdTx{}, errors.Wrap(err, "error encoding SendReq to json")
+	}
+
+	resp, err := http.Post(url, "", bytes.NewBuffer(bz))
+	if err != nil {
+		return types2.StdTx{}, errors.Wrap(err, "error while sending post request")
+	}
+
+	bz, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return types2.StdTx{}, errors.Wrap(err, "error reading SendReq response body")
+	}
+
+	var tx types2.StdTx
+	err = val.ClientCtx.JSONMarshaler.UnmarshalJSON(bz, &tx)
+	if err != nil {
+		return types2.StdTx{}, errors.Wrap(err, "error unmarshaling to StdTx SendReq response")
+	}
```

Comment:
Can we move all of this out to an auxiliary function? See `rest.GetRequest` as a reference.